server/src/main/java/com/defold/extender/services/CocoaPodsService.java

```diff
@@ -0,0 +1,625 @@
+        Set<String> pods = new HashSet<>();
+        while (!lines.isEmpty()) if (lines.remove(0).startsWith("PODS:")) break;
+        while (!lines.isEmpty()) {
+            String line = lines.remove(0);
+            if (line.trim().isEmpty()) break;
+            // - FirebaseCore (8.13.0):
+            if (line.startsWith("  -")) {
+                // '- "GoogleUtilities/Environment (7.10.0)"":'   ->   'GoogleUtilities/Environment (7.10.0)'
+                String pod = line.trim().replace("- ", "").replace(":", "").replace("\"","");
+                pods.add(pod);
+            }
+        }
+
+        File podsDir = new File(workingDir, "Pods");
+        List<PodSpec> specs = new ArrayList<>();
+        Map<String, PodSpec> specsMap = new HashMap<>();
+        for (String pod : pods) {
+            // 'GoogleUtilities/Environment (7.10.0)'  -> 'GoogleUtilities/Environment'
+            String podnameparts[] = pod.replaceFirst(" \\(.*\\)", "").split("/");
+            String mainpodname = podnameparts[0];
+            if (!specsMap.containsKey(mainpodname)) {
+                // 'GoogleUtilities/Environment (7.10.0)'   ->   'GoogleUtilities --version=7.10.0'
+                String args = pod.replaceFirst("/.*?\\s", " ").replace("(", "--version=").replace(")", "");
+                String cmd = "pod spec cat " + args;
+                String specJson = execCommand(cmd).replace(cmd, "");
+
+                specsMap.put(mainpodname, createPodSpec(specJson, podsDir));
+            }
+
+            PodSpec mainpod = specsMap.get(mainpodname);
+            if (podnameparts.length == 1) {
+                specs.add(mainpod);
+            }
+            else {
+                String subspecname = podnameparts[1];
+                for (PodSpec subspec : mainpod.subspecs) {
+                    if (subspec.name.equals(subspecname)) {
+                        specs.add(subspec);
+                        break;
+                    }
+                }
+            }
+        }
+
+        return specs;
+
+    }
+
+    /**
+     * Entry point for Cocoapod dependency resolution.
+     * @param jobDirectory Root directory of the job to resolve
+     * @param platform Which platform to resolve pods for
+     * @return ResolvedPods instance with list of pods, install directory etc
+     */
+    public ResolvedPods resolveDependencies(File jobDirectory, String platform) throws IOException, ExtenderException {
+        if (!platform.contains("ios") && !platform.contains("osx")) {
+            throw new ExtenderException("Unsupported platform " + platform);
+        }
+
+        long methodStart = System.currentTimeMillis();
+        LOGGER.info("Resolving Cocoapod dependencies");
+
+        File workingDir = new File(jobDirectory, "CocoaPodsService");
+        File frameworksDir = new File(workingDir, "frameworks");
+        workingDir.mkdirs();
+
+        String platformVersion = createMainPodFile(jobDirectory, workingDir, platform);
```

Comment:
Resolving dependencies is done in steps:

1. Create a Podfile containing all dependencies collected from uploaded extensions
2. Install pods using the Cocoapods command line tool
3. Parse the installed pods
   * Parse Podfile.lock containing list of installed pods
   * Get the Podspec for each pod (json)
   * Create a PodSpec java class instance for each pod
4. Copy .xcframeworks from installed pods

Reply:
wow, a lot of nuances esp. with Podfile.lock

it would be interesting to make a stress test (actually the most useful usecase for the feature) with ADS mediation:

```
pod 'IronSourceSDK','7.2.7.0'
pod 'IronSourceAdColonyAdapter','4.3.15.4'
pod 'IronSourceAdMobAdapter','4.3.40.0'
pod 'IronSourceAppLovinAdapter','4.3.37.1'
pod 'IronSourceAPSAdapter','4.3.6.0'
pod 'IronSourceChartboostAdapter','4.3.12.4'
pod 'IronSourceCSJAdapter', '4.3.1.0'
pod 'IronSourceFacebookAdapter','4.3.38.4'
pod 'IronSourceFyberAdapter','4.3.27.0'
pod 'IronSourceHyprMXAdapter','4.3.1.4'
pod 'IronSourceInMobiAdapter','4.3.16.0'
pod 'IronSourceLiftoffAdapter','4.3.3.4'
pod 'IronSourceMaioAdapter','4.1.10.18'
pod 'IronSourceMintegralAdapter', '4.3.12.0'
pod 'IronSourceMyTargetAdapter','4.1.15.2'
pod 'IronSourcePangleAdapter','4.3.18.0'
pod 'IronSourceSmaatoAdapter','4.3.9.0'
pod 'IronSourceSuperAwesomeAdapter','4.1.5.2'
pod 'IronSourceTapjoyAdapter','4.1.23.1'
pod 'IronSourceTencentAdapter','4.3.3.9'
pod 'IronSourceUnityAdsAdapter','4.3.25.2'
pod 'IronSourceVungleAdapter','4.3.24.0'
pod 'IronSourceYahooAdapter','4.3.4.1'
```
(I took it from here https://developers.is.com/ironsource-mobile/ios/mediation-networks-ios/#step-1)

Reply:
The above resolves and builds, except for IronSourceYahooAdapter which contains swift code